sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -86,67 +84,98 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
     this.mapboxRenderer = mapboxRenderer
     this.renderHandlerThread = handlerThread
     this.eglCore = eglCore
+    this.eglSurface = eglCore.eglNoSurface
   }
 
-  private fun postPrepareRenderFrame() {
-    renderHandlerThread.post { prepareRenderFrame() }
+  private fun postPrepareRenderFrame(delayMillis: Long = 0L) {
+    renderHandlerThread.postDelayed(
+      {
+        prepareRenderFrame()
+      },
+      delayMillis
+    )
   }
 
-  private fun checkSurfaceReady(creatingSurface: Boolean): Boolean {
+  private fun isValidRenderingEnvironment(creatingSurface: Boolean): Boolean {
     lock.withLock {
       try {
-        surface?.let {
-          if (eglSurface == null || eglSurface == EGL10.EGL_NO_SURFACE) {
-            return prepareEglSurface(it, creatingSurface)
+        val eglConfigOk = checkEglConfig()
+        val androidSurfaceOk = checkAndroidSurface()
+        if (eglConfigOk && androidSurfaceOk) {
+          // on Android SDK <= 23 at least on x86 emulators we need to force set EGL10.EGL_NO_CONTEXT
+          // when resuming activity
+          if (creatingSurface) {
+            eglCore.makeNothingCurrent()
+          }
+          // it's safe to use !! here as we checked surface above
+          val eglSurfaceOk = checkEglSurface(surface!!)
+          if (eglSurfaceOk) {
+            val eglContextOk = checkEglContextCurrent()
+            // finally we can create native renderer if needed or just report OK
+            if (eglContextOk) {
+              if (!nativeRenderCreated) {
+                mapboxRenderer.onSurfaceCreated()
+                mapboxRenderer.onSurfaceChanged(
```

Comment:
nit : may just set `surfaceSizeChanged` to true - will be set later in `checkSurfaceSizeChanged`

Reply:
doesn't change anything I guess, could leave as is